src/mds/Server.cc

```diff
@@ -5981,6 +5981,49 @@ int Server::parse_layout_vxattr(string name, string value, const OSDMap& osdmap,
   return 0;
 }
 
+int64_t Server::make_quota_max_byte_parseable(string val)
+{
+  string before_decimal;
+  string after_decimal;
+  int64_t dec_len;
+  int64_t q;
+  string cast_err;
+  
+  size_t dec_index = val.find_first_of(".");
+  size_t unit_index = val.find_first_not_of("0123456789-+.");
```

Comment:
Why do we have `-+` part of "not of"? `-` or `+` are invalid chars and I think shouldn't be ignored.